### PR TITLE
fix: Explicitly set `KATIB_DB_MANAGER_SERVICE_PORT`

### PR DESCRIPTION
**Pull request by orfeas-k**
* fix: Explicitly set `KATIB_DB_MANAGER_SERVICE_PORT` in katib-controller
* katib-db-manager: Remove port config option

Closes canonical/bundle-kubeflow#893, canonical/katib-operators#108
Addressed also part of https://github.com/canonical/katib-operators/issues/184

### CODE REVIEW

File: charms/katib-db-manager/config.yaml [hunk 1]
```diff
@@ -1,5 +1 @@
-options:
-  port:
-    type: int
-    default: 6789
-    description: API port
+options: {}
```
Comment (NohaIhab) [inline charms/katib-db-manager/config.yaml:1]: afaik, we can remove the file now

File: charms/katib-controller/src/charm.py [hunk 2]
```diff
@@ -253,7 +253,10 @@ def set_pod_spec(self, event):
                                 "containerPort": self.model.config["metrics-port"],
                             },
                         ],
-                        "envConfig": {"KATIB_CORE_NAMESPACE": self.model.name},
+                        "envConfig": {
+                            "KATIB_CORE_NAMESPACE": self.model.name,
+                            "KATIB_DB_MANAGER_SERVICE_PORT": "6789",
```
Comment (NohaIhab) [inline charms/katib-controller/src/charm.py:258]: comment here to explain why it is hardcoded? maybe refer to the constant in db-manager's `charm.py`

File: charms/katib-db-manager/src/charm.py [hunk 3]
```diff
@@ -23,6 +23,7 @@
 ]
 MYSQL_WARNING = "Relation mysql is deprecated."
 UNBLOCK_MESSAGE = "Remove deprecated mysql relation to unblock."
+SERVICE_PORT = "6789"
```
Comment (NohaIhab) [inline charms/katib-db-manager/src/charm.py:26]: comment with where it is hardcoded in upstream?

**Review by NohaIhab — COMMENTED** [review]
thx @orfeas-k 
I have tested this using the charms from the PR and it resolves the bug!
left some small comments

File: charms/katib-db-manager/src/charm.py (hunk 3 above)
Comment (NohaIhab) [reply charms/katib-db-manager/src/charm.py:28]: can you link the 0.16 version instead?

Comment (orfeas-k) [reply charms/katib-db-manager/src/charm.py:28]: added

**Pull request merged**